sdk/storage/src/cloud_location.rs

```diff
@@ -44,28 +88,123 @@ impl CloudLocation {
+                            account: account.clone(),
+                            credentials: credentials.clone(),
+                        }
+                        .url(service_type),
+                        AZURE_US_GOV => CloudLocation::USGov {
+                            account: account.clone(),
+                            credentials: credentials.clone(),
+                        }
+                        .url(service_type),
+                        AZURE_CHINA_CLOUD => CloudLocation::China {
+                            account: account.clone(),
+                            credentials: credentials.clone(),
+                        }
+                        .url(service_type),
+                        _ => {
+                            return Err(azure_core::Error::with_message(
+                                azure_core::error::ErrorKind::Other,
+                                || {
+                                    format!(
+                                        "Auto-detect encountered an invalid cloud name, allowed values are: {AZURE_CLOUD}, {AZURE_PUBLIC_CLOUD}, {AZURE_US_GOV}, {AZURE_CHINA_CLOUD}.",
+                                    )
+                                },
+                            ));
+                        }
+                    };
+                } else {
+                    return Err(azure_core::Error::with_message(
+                        azure_core::error::ErrorKind::Other,
+                        || {
+                            "Auto-detect could not find a cloud name from the current environment."
+                                .to_string()
+                        },
+                    ));
+                }
+            }
         };
         Ok(url::Url::parse(&url)?)
     }
 
+    /// Returns the storage credentials for this cloud location,
+    ///
     pub fn credentials(&self) -> &StorageCredentials {
         match self {
-            CloudLocation::Public { credentials, .. } => credentials,
-            CloudLocation::China { credentials, .. } => credentials,
+            CloudLocation::Public { credentials, .. }
+            | CloudLocation::China { credentials, .. }
+            | CloudLocation::USGov { credentials, .. }
+            | CloudLocation::Custom { credentials, .. }
+            | CloudLocation::AutoDetect { credentials, .. } => credentials,
             CloudLocation::Emulator { .. } => &EMULATOR_CREDENTIALS,
-            CloudLocation::Custom { credentials, .. } => credentials,
+        }
+    }
+
+    /// Finds the cloud name, first by environment variable, then by parsing the current user's $HOME/.azure/config file
+    ///
+    fn find_cloud_name() -> Option<String> {
+        if let Ok(name) = std::env::var("AZURE_CLOUD_NAME") {
+            Some(name)
+        } else if let Ok(home_dir) = std::env::var("HOME") {
+            if let Some(config) = PathBuf::from(home_dir)
+                .join(".azure/config")
```

Comment:
I'm not sure the storage crate should be reading `$HOME/.azure/config`. Is something like this done in the other Azure SDKs?

Reply:
Can't find any other usage besides the cli repos. It appears the storage sdk for .net handles this by using an 'EndpointSuffix=' arg in the connection string. Is there a possible middle ground of moving this to a shared location or what are your thoughts?